fix: port various changes from weasel environment repository

ccojocar opened the pull request:
Only the relevant changes are moved over. These are changes which are not specific to
jenkins-x production environment (environment-tekton-weasel-dev).

fixes jenkins-x/jx#5190
fixes jenkins-x/jx#4865

ccojocar committed:
fix: port various changes from weasel environment repository

Only the relevant changes are moved over. These are changes which are not specific to
jenkins-x production environment (environment-tekton-weasel-dev).

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: remove the execute premission from some files

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: remove md file to get the step helm apply working

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

cagiti commented on env/templates/700-hook-ing.yaml:
```diff
@@ -5,6 +5,11 @@ metadata:
+    certmanager.k8s.io/issuer: letsencrypt-prod
+{{- else }}
+    certmanager.k8s.io/issuer: letsencrypt-staging
+{{- end }}
```
This if/else/end block is no longer required within the ingress

ccojocar replied:
It seems to be present on https://github.com/jenkins-x/environment-tekton-weasel-dev/blob/676ef0e92724ed89ae6193aa5a7ac20a2b01b1f8/env/templates/700-hook-ing.yaml#L8. 

Should be removed from all ingress resources? Can we use only `certmanager.k8s.io/issuer: letsencrypt-prod`?

cagiti replied:
Yes, should not be removed... I don't think the PR that James R raised on the boot config was merged before the weasel environment was created

rawlingsj replied:
yeah that block can be removed as we use external dns and the new issuer format https://github.com/jenkins-x/environment-tekton-weasel-dev/blob/676ef0e92724ed89ae6193aa5a7ac20a2b01b1f8/env/templates/cert-manager-prod-issuer.yaml#L13-L26 this means annotations on ingress rules aren't needed.

ccojocar replied:
Should I remove it on all ingress resources?

cagiti replied:
yes, if it's present 👍

ccojocar replied:
It was only on hook. All ingress should not have now any cert-manager annotations.

cagiti commented on env/parameters.schema.json:
```diff
@@ -40,11 +39,6 @@
           "title": "Pipeline bot Git username",
           "description": "The Git user that will perform git operations inside a pipeline. It should be a user within the Git organisation/owner where environment repositories will live. This is normally a bot."
         },
-        "email": {
```
Looks like some changes are being reverted, possibly just needs rebasing.

ccojocar replied:
I am not sure, but it seems to be removed also form here https://github.com/jenkins-x/environment-tekton-weasel-dev/blob/676ef0e92724ed89ae6193aa5a7ac20a2b01b1f8/env/parameters.schema.json#L31

Was this accidentally done?

cagiti replied:
I think the weasel environment was created before this PR was merged https://github.com/jenkins-x/jenkins-x-boot-config/pull/44/files

ccojocar replied:
I reverted the changes to include the `email` field.

ccojocar commented on jenkins-x.yml:
```diff
@@ -19,43 +19,51 @@ pipelineConfig:
                 args: ['step','create','install', 'values', '-b']
-              - name: install-vault
-                dir: /workspace/source/systems/vault
+              - name: install-external-dns
```
@cagiti This step needs to be conditionally executed only when `externalDNS` is enabled. The BDD tests are currently failing when the externalDNS is disabled. I am looking to find a solution for this.

cagiti replied:
yeah, would be good to perhaps wrap the contents of the files that step executes with a condition so if you haven't got it enables it basically no-ops

ccojocar replied:
Any idea how is the best to do this. The values is defined in `jx-requirements.yml` and I don't see how the values are propagated into the `jenkins-x.yml` pipeline file. 

It seems that some commands are loading the `jx-requirements.yml` file and are parsing a specific key from it. e.g `step boot vault` is looking for `secretStore: vault`.

ccojocar committed:
chore: revert the parameters.schema.json to include the email field

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: remove the cert-manger annotations on hook ingress resource

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
fix: disable the install external dns step if it is disabled in the jx-requirements.yml

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented on jenkins-x.yml:
```diff
@@ -19,43 +19,50 @@ pipelineConfig:
                   - name: DEPLOY_NAMESPACE
                     value: cert-manager
-              - name: cert-manager
+              - name: install-cert-manager
```
also this step needs to be excluded when tls is disabled

ccojocar committed:
fix: move the acme resources in a template folder and remove the condition on external DNS

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

abayer commented:
The `jenkins-x.yml` was busted, but I've just now fixed it.

abayer commented:
The failures now seem legit and aren't due to `jenkins-x.yml` format issues. =)

ccojocar commented:
> The failures now seem legit and aren't due to jenkins-x.yml format issues. =)

yeah, that was mistakenly removed but still fails because of the empty charts (e.g. when cert-manager and external-dns are disabled). 

This fix is required in the builder-go https://github.com/jenkins-x/jx/pull/5223.

ccojocar committed:
fix: disable the optional charts installation

Disable the installation of the cert-manager, external-dns and acme charts when
they are disable in the jx-requirements.yml file.

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
The `builder-go` still needs to be updated. I am waiting to be released with the latest fixes from jx.

ccojocar committed:
chore: remove the external-dns secret from templates

This secret gets created dynamically by boot.

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: update the builder-go in the pipelines to version 0.1.686

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test all

ccojocar committed:
chore: use the builder-go image 0.1.686

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test all

ccojocar committed:
chore: add the versions stream url into jx-requirements file

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test bdd-vault

ccojocar commented:
/test bdd-vault

ccojocar commented:
/test bdd-vault
